spark/src/test/java/com/netflix/iceberg/spark/data/TestHelpers.java

```diff
@@ -280,4 +284,289 @@ private static void assertEqualsUnsafe(Type type, Object expected, Object actual
         throw new IllegalArgumentException("Not a supported type: " + type);
     }
   }
+
+  /**
+   * Check that the given InternalRow is equivalent to the Row.
+   * @param prefix context for error messages
+   * @param type the type of the row
+   * @param expected the expected value of the row
+   * @param actual the actual value of the row
+   */
+  public static void assertEquals(String prefix, Types.StructType type,
+                                  InternalRow expected, Row actual) {
+    if (expected == null || actual == null) {
+      Assert.assertEquals(prefix, expected, actual);
+    } else {
+      List<Types.NestedField> fields = type.fields();
+      for (int c = 0; c < fields.size(); ++c) {
+        String fieldName = fields.get(c).name();
+        Type childType = fields.get(c).type();
+        switch (childType.typeId()) {
+          case BOOLEAN:
+          case INTEGER:
+          case LONG:
+          case STRING:
+          case DECIMAL:
+          case DATE:
+          case TIMESTAMP:
+            Assert.assertEquals(prefix + "." + fieldName + " - " + childType,
+                getPrimativeValue(expected, c, childType),
+                getPrimativeValue(actual, c, childType));
+            break;
+          case FLOAT:
+          case DOUBLE:
+            assertEqualsDouble(prefix + "." + fieldName,
+                (Double) getPrimativeValue(expected, c, childType),
+                (Double) getPrimativeValue(actual, c, childType));
+            break;
+          case UUID:
+          case FIXED:
+          case BINARY:
+            assertEqualBytes(prefix + "." + fieldName,
+                (byte[]) getPrimativeValue(expected, c, childType),
+                (byte[]) actual.get(c));
+            break;
+          case STRUCT: {
+            Types.StructType st = (Types.StructType) childType;
+            assertEquals(prefix + "." + fieldName, st,
+                expected.getStruct(c, st.fields().size()), actual.getStruct(c));
+            break;
+          }
+          case LIST:
+            compareLists(prefix + "." + fieldName, childType.asListType(),
```

Comment:
I think I prefer the `assert` naming convention to `compare` because it is clear that the result when something is different is an AssertionError.

Reply:
ok